platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/status/DefaultStatusStateMachine.java

```diff
@@ -83,19 +71,12 @@ public class PlatformStatusStateMachine implements PlatformStatusGetter {
     /**
      * Constructor
      *
-     * @param time                 a source of time
-     * @param config               the platform status config
-     * @param statusChangeConsumer consumes any status changes
+     * @param context the platform context
      */
-    public PlatformStatusStateMachine(
-            @NonNull final Time time,
-            @NonNull final PlatformStatusConfig config,
-            @NonNull final Consumer<PlatformStatus> statusChangeConsumer) {
-
-        this.time = Objects.requireNonNull(time);
-        this.statusChangeConsumer = Objects.requireNonNull(statusChangeConsumer);
-        this.currentStatusLogic = new StartingUpStatusLogic(config);
-        this.currentStatus = new AtomicReference<>(currentStatusLogic.getStatus());
+    public DefaultStatusStateMachine(@NonNull final PlatformContext context) {
+        this.time = Objects.requireNonNull(context.getTime());
```

Comment:
Nit, we probably don't need to verify that the context is returning non-null time (although it doesn't hurt to do so I guess).

Reply:
fixed